src/main/java/com/solvd/pages/ResultPage.java

```diff
@@ -1,14 +1,16 @@
 package com.solvd.pages;
 
+import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import org.openqa.selenium.support.PageFactory;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
 
+import java.time.Duration;
 import java.util.List;
 
-public class ResultPage {
-    private WebDriver driver;
+public class ResultPage extends AbstractPage {
 
     @FindBy(xpath = "//span[@class ='a-size-medium a-color-base a-text-normal']")
```

Comment:
This xpath looks a bit unstable to me as one day size or text may change from medium/normal to large, i checked on amazon website that you can use ".//div[@class='a-section']" to count number of results or ".//div[@class='a-section']//h2" for number of headings in results

Reply:
thanks. it works better